pkg/controller/ephemeraljob/ephemeraljob_controller.go

```diff
@@ -375,3 +388,20 @@ func (r *ReconcileEphemeralJob) updateJobStatus(job *appsv1alpha1.EphemeralJob)
 	klog.V(5).Infof("Updating job %s status %#v", job.Name, job.Status)
 	return r.Status().Update(context.TODO(), job)
 }
+
+func (r *ReconcileEphemeralJob) removeEphemeralContainers(job *appsv1alpha1.EphemeralJob) error {
+	targetPods, err := r.filterPods(job)
+	if err != nil {
+		klog.Errorf("Failed to get ephemeral job %s/%s related target pods: %v", job.Namespace, job.Name, err)
+		return err
+	}
+
+	var errors error
+	for _, pod := range targetPods {
+		if e := econtainer.New(job).RemoveEphemeralContainer(pod); e != nil {
```

Comment:
`RemoveEphemeralContainer` should not get the single pod as parameter. If the pod has ephemeral containers that owned by multiple jobs, you should not remove them all in this place.